### ISSUE TITLE
Allow to combine object shorthand notation with type assertion

### ISSUE THREAD

**Issue by danielrentz** [microsoft/TypeScript#53231]
# Suggestion

## 🔍 Search Terms

object shorthand notation type assertion

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Allow to combine object shorthand notation with type conversions.

## 📃 Motivating Example

Using external libraries that are not typed very well.

```ts
function myFunc(type: string, data: unknown) {
  if (someExternalFunction(data)) {
    // here it's known that `data` is a string
    someOtherFunction({ type, data: data as string });
  }
}
```

It would be nice to be able to shorten this to
```ts
    someOtherFunction({ type, data as string });
```
or
```ts
    someOtherFunction({ type, <string>data });
```

## 💻 Use Cases

Just to make source code more concise.

**Comment by MartinJohns** [comment]
Last time this suggestion was closed due to lack of feedback and a comment to stay "well-clear of this syntactic space": https://github.com/microsoft/TypeScript/issues/13035#issuecomment-867859066

> I think we want to stay well-clear of this syntactic space to keep room for future TC39.

**Comment by danielrentz** [comment]
Ok... The issue search function did not bring that thing up. (as usual when I try to use the search :) )
